src/Resource.php

```diff
@@ -231,10 +236,9 @@ public function query()
         if (request('search')) {
             Search::apply($query, request('search'), static::$search);
         }
-
         if (static::$sortable) {
             Sort::apply($query, static::$sortField, 'ASC');
-        } elseif (request('sort')) {
+        } elseif (request('sort') && in_array(request('sort'), $this->getValidSortFields()->toArray())) {
```

Comment:
mmm potser ho faria diferent, aquest $validSortFields no els posaria aqui, ja que el resource ja the la variable $sort
per tant ho faria directament amb aquesta variable `in_array(request('sort'), $this->sort)`

Reply:
mmm no té cap variable que es digui sort

Reply:
Tens rao, m'he confos amb el search,
llavors el que hauriem de fer es agafar els fields i filtrar-los per els que tinguin stinguin $sortable == true i llavors agafar-ne el camp $field 

es a dir

`$this->fields()->where('sortable')->pluck('field')->contains(request('sort'));`

o algu així